src_c/geometry.c

```diff
@@ -8,6 +8,106 @@
 
 #define PYGAMEAPI_GEOMETRY_NUMSLOTS 21
 
+static PyObject *
+pg_raycast(PyObject *_null, PyObject *const *args, Py_ssize_t nargs)
+{
+    PyObject **farr;
+    Py_ssize_t farr_length;
+    Py_ssize_t loop;
+    double max_t;
+    pgLineBase line;
+
+    if (nargs == 4) {
+        if (!pg_TwoDoublesFromObj(args[0], &line.x1, &line.y1)) {
+            return RAISE(PyExc_TypeError,
+                         "the origin requires a pair of floats");
+        }
+        double max_dist;
+        if (!pg_DoubleFromObj(args[2], &max_dist)) {
+            return RAISE(PyExc_TypeError, "invalid max distance");
+        }
+        if (max_dist < 0) {
+            max_dist = DBL_MAX;
+        }
+        if (!pg_TwoDoublesFromObj(args[1], &line.x2, &line.y2)) {
+            double angle;
+            if (!pg_DoubleFromObj(args[1], &angle)) {
+                return RAISE(PyExc_TypeError,
+                             "expected a pair of floats or a single float");
+            }
+
+            if (max_dist != DBL_MAX) {
+                line.x2 = line.x1 - cos(angle * PI / 180) * max_dist;
+                line.y2 = line.y1 - sin(angle * PI / 180) * max_dist;
+            }
+            else {
+                line.x2 = line.x1 - cos(angle * PI / 180);
+                line.y2 = line.y1 - sin(angle * PI / 180);
+            }
```

Comment:
All of this you could put inside the if at line 29, but the thing is that you should extract the `max_dist `value AFTER the angle/end_pos value. in this way code is more compact and organized.